File: dev/ios/GalleryAPI/galleryapi/GalleryAPI.m
//
//  GalleryAPI.m
//  galleryapi
//
//  Created by Thomas Decaux on 26/02/2015.
//  Copyright (c) 2015 SubitoLabs. All rights reserved.
//

#import "GalleryAPI.h"

@interface GalleryAPI ()

@end

@implementation GalleryAPI

- (void) getAlbums:(void (^) (NSArray *))successHandler
{
    __block ALAssetsLibrary *library = [[ALAssetsLibrary alloc] init];
    __block NSMutableArray *albums = [[NSMutableArray alloc] init];
    
    [library enumerateGroupsWithTypes:ALAssetsGroupAll usingBlock:^(ALAssetsGroup *group, BOOL *stop)
     {
         if (group == nil)
         {
             successHandler(albums);
         }
         else
         {
             [albums addObject:@{
                                 @"title" : [group valueForProperty:ALAssetsGroupPropertyName],
                                 @"assets" : [NSString stringWithFormat:@"%ld", group.numberOfAssets]
                                 }];
         }
         
     } failureBlock: ^(NSError *error) {
         // Typically you should handle an error more gracefully than this.
         NSLog(@"No groups");
     }];
}

- (void) getAlbumAssets:(NSString*) album withSuccessHandler:(void (^) (NSArray *))successHandler
{
    __block ALAssetsLibrary *library = [[ALAssetsLibrary alloc] init];
    __block NSMutableArray *assets = [[NSMutableArray alloc] init];
    
    [library enumerateGroupsWithTypes:ALAssetsGroupAll usingBlock:^(ALAssetsGroup *group, BOOL *stop)
     {
         if (group == nil)
         {
             successHandler(assets);
         }
         else
         {
             if (album == [group valueForProperty:ALAssetsGroupPropertyName])
             {
                 [group enumerateAssetsWithOptions:NSEnumerationReverse usingBlock:
                  ^(ALAsset *result, NSUInteger index, BOOL *stop)
                  {
                      if(result == nil)
                      {
                          successHandler(assets);
                      }
                      else
                      {
                          ALAssetRepresentation *representation = result.defaultRepresentation;
                          
                          [assets addObject:@{
                                              @"title" : representation.filename,
                                              }];
                          
                      }
                  }
                  ];
             }
             
         }
     }
                         failureBlock: ^(NSError *error) {
                             // Typically you should handle an error more gracefully than this.
                             NSLog(@"No groups");
                         }];
}

@end

File: dev/ios/GalleryAPI/GalleryAPI.swift
//
//  GalleryAPI.swift
//  GalleryAPI
//
//  Created by Thomas Decaux on 08/02/2015.
//  Copyright (c) 2015 SubitoLabs. All rights reserved.
//

import Foundation
import AssetsLibrary

class GalleryAPI
{
    func getAlbums() -> NSArray
    {
        var albums : NSMutableArray = NSMutableArray()
        var library = ALAssetsLibrary()
        
        library.enumerateGroupsWithTypes(ALAssetsGroupAll, usingBlock: { (group, stop) -> Void in
           
            var title : NSString = group.valueForProperty(ALAssetsGroupPropertyName) as String
            
            albums.addObject(["title" : title])
            
            }) { (error) -> Void in
                println("problem loading albums: \(error)")
        }
        
        return albums
    }
    
    func getMedia(album : NSString)
    {
        
    }
}